pkg/dependenciesdistributor/dependencies_distributor.go

```diff
@@ -228,7 +163,7 @@ func dependentObjectReferenceMatches(objectKey keys.ClusterWideKey, referenceBin
 	}
 
 	for _, dependence := range dependenciesSlice {
-		if objectKey.Version == dependence.APIVersion &&
+		if objectKey.GroupVersion().String() == dependence.APIVersion &&
```

Comment:
> This test is failing:
https://github.com/karmada-io/karmada/actions/runs/5369361839/jobs/9740779864?pr=3614

Because of this line of code.

 I added unit tests for it. 

/cc @RainbowMango